### PR TITLE
Add locator sanity tests

### PR DESCRIPTION
**Pull request by inancgumus**
Moves the strict tests into the sanity checks tests along with the timeout tests.

The reason for using a `timeout` closure in the test is to use different goja runtimes because of goja races. The timeout tests should be done in a new browser, otherwise unpredictable errors happen. The side benefit is that they run in parallel.

**Note:** Since #413 is based on this branch, please send your request changes or suggestions to #413.

This PR finishes the "Check for empty matches" task in #366.

### CODE REVIEW

**Review by imiric — CHANGES_REQUESTED** [review]
LGTM, just minor suggestions.

I'm not a fan of grouping tests by type in this way, even if it avoids some repetition. Tests should be grouped per method or feature, otherwise now we have to remember to add tests in both `TestLocator*` and in this new general `TestLocatorSanity` test.

File: tests/locator_test.go [hunk 1]
```diff
@@ -23,14 +25,8 @@ func TestLocatorClick(t *testing.T) {
...
-	t.Run("ok", func(t *testing.T) {
-		p.Locator("#link", nil).Click(nil)
-		require.True(t, tb.asGojaBool(p.Evaluate(tb.toGojaValue(`() => window.result`))), "could not click the link")
-	})
-	// The strict mode should disallow selecting multiple elements.
-	t.Run("strict", func(t *testing.T) {
-		require.Panics(t, func() { p.Locator("a", nil).Click(nil) })
-	})
+	p.Locator("#link", nil).Click(nil)
+	require.True(t, tb.asGojaBool(p.Evaluate(tb.toGojaValue(`() => window.result`))), "could not click the link")
```
[the remark this answers is not in the document]
Comment (inancgumus) [reply tests/locator_test.go:29]: Please see #413 and let me know which ones should I change.

[the remark this answers is not in the document]
Comment (inancgumus) [reply tests/locator_test.go:29]: Please see https://github.com/grafana/xk6-browser/pull/413/commits/cfcd36f213d627f1ce7c29acd0ec27c8afa79bae in #413. Sorry to spread the work to two PRs.

File: tests/locator_test.go [hunk 2]
```diff
@@ -434,7 +312,111 @@ func TestLocatorWaitFor(t *testing.T) {
...
+			"WaitFor", func(l api.Locator, tb *testBrowser) { l.WaitFor(timeout(tb)) },
+		},
+		{
+			"DispatchEvent", func(l api.Locator, tb *testBrowser) {
+				l.DispatchEvent("click", tb.toGojaValue("mouseevent"), timeout(tb))
+			},
+		},
+		{
+			"Tap", func(l api.Locator, tb *testBrowser) { l.Tap(timeout(tb)) },
+		},
```
[the remark this answers is not in the document]
Comment (inancgumus) [reply tests/locator_test.go:338]: Please see https://github.com/grafana/xk6-browser/pull/413/commits/cfcd36f213d627f1ce7c29acd0ec27c8afa79bae.

**Comment by imiric** [comment]
> I've moved the tests in the same place https://github.com/grafana/xk6-browser/commit/cfcd36f213d627f1ce7c29acd0ec27c8afa79bae. How does it look now?

Now they're closer, so slightly better :smile: But it's fine, we can work with that.

**Review by imiric — APPROVED** [review]

**Pull request merged**